src/send/SelectRecipientButtons.tsx

```diff
@@ -0,0 +1,177 @@
+import React, { useState } from 'react'
+import { useAsync } from 'react-async-hook'
+import { useTranslation } from 'react-i18next'
+import { Platform } from 'react-native'
+import { PERMISSIONS, PermissionStatus, RESULTS, check, request } from 'react-native-permissions'
+import { SendEvents } from 'src/analytics/Events'
+import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
+import { phoneNumberVerifiedSelector } from 'src/app/selectors'
+import Dialog from 'src/components/Dialog'
+import SelectRecipientButton from 'src/components/SelectRecipientButton'
+import QRCode from 'src/icons/QRCode'
+import Social from 'src/icons/Social'
+import { navigate } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import useSelector from 'src/redux/useSelector'
+import Logger from 'src/utils/Logger'
+import { navigateToPhoneSettings } from 'src/utils/linking'
+
+const CONTACTS_PERMISSION =
+  Platform.OS === 'ios' ? PERMISSIONS.IOS.CONTACTS : PERMISSIONS.ANDROID.READ_CONTACTS
+
+type Props = {
+  onContactsPermissionGranted: () => void
+}
+
+export default function SelectRecipientButtons({ onContactsPermissionGranted }: Props) {
+  const { t } = useTranslation()
+  const phoneNumberVerified = useSelector(phoneNumberVerifiedSelector)
+  const [contactsPermissionStatus, setContactsPermissionStatus] = useState<
+    PermissionStatus | undefined
+  >(undefined)
+  const [showConnectPhoneNumberModal, setShowConnectPhoneNumberModal] = useState(false)
+  const [showEnableContactsModal, setShowEnableContactsModal] = useState(false)
+  const [navigateToPhoneVerification, setNavigateToPhoneVerification] = useState(false)
+
+  useAsync(async () => {
+    if (!contactsPermissionStatus) {
+      setContactsPermissionStatus(await check(CONTACTS_PERMISSION))
```

Comment:
the reason I made this a state because this can change during the request flow. I missed setting the state after a request completes. Updated now.

Reply:
> the reason I made this a state because this can change during the request flow.

I'm not sure that i understand this part, how can this state change during the flow? does the user have to manually close the app -> go to settings -> adjust permissions -> come back to the app? (in this case, i don't think having a state variable helps to trigger a re-check of the permission?). maybe i'm missing something, but the component looks like it will only check and update the value of `contactsPermissionStatus` when the user presses on the contacts button or if it was initially undefined (through the useAsync)...i'm missing the link for how this state variable is effective at tracking the permission changes after the useAsync runs for the first time. is it not fine to simplify the assumption to the user does not leave this flow to update their permissions? in that case we could just check for permission on component mount only (with an empty dependency array)?

more importantly, i think it is discouraged that the async function produces side effects - any state changes should be handled in the `onSuccess` callback

Reply:
the state can change during the flow when the native modal is shown the first time. The state can change from denied -> granted / blocked. Once it gets to blocked, the state can change only outside the app through settings. I updated the useAsync to use the `onSuccess` callback for state updates.